linera-execution/src/system.rs

```diff
@@ -35,6 +35,39 @@ use thiserror::Error;
 #[cfg(any(test, feature = "test"))]
 use {crate::applications::ApplicationRegistry, std::collections::BTreeSet};
 
+#[derive(Clone, Debug, Default, Eq, Hash, PartialEq)]
+pub struct Committees(pub BTreeMap<Epoch, Committee>);
```

Comment:
Same here: just (de)serialize `Epoch` as a `u64` in the case of readable formats.

Reply:
Something like that:
```
impl Serialize for Epoch {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: serde::ser::Serializer,
    {
        if serializer.is_human_readable() {
            serializer.serialize_u64(&self.0)
        } else {
            serializer.serialize_newtype_struct("Epoch", &self.0)
        }
    }
}

impl<'de> Deserialize<'de> for Epoch {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: serde::de::Deserializer<'de>,
    {
        if deserializer.is_human_readable() {
            let s = u64::deserialize(deserializer)?;
            let value = Self::from_str(&s).map_err(serde::de::Error::custom)?;
            Ok(value)
        } else {
            #[derive(Deserialize)]
            #[serde(rename = "Epoch")]
            struct E(u64);

            let value = E::deserialize(deserializer)?;
            Ok(Self(value.0))
        }
    }
}
```